docs/architecture/adr-100-data-companion-push-api.md

```diff
@@ -0,0 +1,441 @@
+not lose any critical data, we need to impose some kind of limit on the number
+of requests we will buffer before crashing the Tendermint node. In case there is
+manual intervention needed to restore the companion service, Tendermint should
+attempt to flush this buffer upon startup before continuing normal operations.
+
+### Configuration
+
+The following configuration file update is proposed to support the data
+companion API.
+
+```toml
+[data_companion]
+# By default, the data companion service interaction is disabled. It is
+# recommended that this only be enabled on full nodes and not validators so as
+# to minimize the possibility of network instability.
+enabled = false
+
+# Address at which the gRPC companion service server is hosted. It is
+# recommended that this companion service be co-located at least within the same
+# data center as the Tendermint node to reduce the risk of network latencies
+# interfering in node operation.
+addr = "http://localhost:26659"
+
+# Use the experimental gzip compression call option when submitting data to the
+# server. See https://pkg.go.dev/google.golang.org/grpc#UseCompressor
+experimental_use_gzip = false
+
+# Controls the BlockCommitted gRPC call.
+[data_companion.block_committed]
+# Enable the BlockCommitted gRPC call. Only relevant if the data companion is
+# enabled.
+enabled = true
+# Additional fields to publish in each BlockCommittedRequest sent to the
+# companion. Available options:
+# - "finalize_block_response": Also publish the FinalizeBlock response related
+#                              to the block in the BlockCommittedRequest.
+additionally_publish = ["finalize_block_response"]
+```
+
+It is unclear at present whether compressing the data being sent to the
+companion service will result in meaningful benefits.
+
+### Monitoring
+
+To monitor the health of the interaction between Tendermint and the companion
+service, the following additional Prometheus metrics are proposed:
+
+- `data_companion_send_time` - A gauge that indicates the maximum time, in
+  milliseconds, taken to send a single request to the companion service take
+  from a rolling window of tracked send times (e.g. the maximum send time over
+  the past minute).
+- `data_companion_send_bytes` - A counter indicating the number of bytes sent to
+  the data companion since node startup.
+- `data_companion_buffer_size` - A gauge indicating the number of requests
+  currently being buffered on disk on behalf of the companion service for each
+  kind of request.
+- `data_companion_buffer_size_bytes` - A gauge indicating how much data is being
+  buffered by Tendermint on disk, in bytes, on behalf of the companion service,
+  for each kind of request.
+
+## Implications
+
+1. We will be able to mark the following RPC APIs as deprecated and schedule
+   them for removal in a future release:
+   1. The [WebSocket subscription API][websocket-api]
+   2. [`/tx_search`]
+   3. [`/block_search`]
+   4. [`/block_results`]
+
```

Comment:
How about 
`/block`
`block_byhash`
`blockchain`
`tx`

Reply:
I'm not sure whether we can remove these yet. I was trying to keep the RPC endpoint removals to the bare minimum here. We can definitely consider removing more endpoints over time as we provide more ways for integrators to access this data through data companions.